lib/sourcemapScanner.js

```diff
@@ -0,0 +1,74 @@
+class SourceMapScannerIterator {
+    constructor(mappings) {
+        this.mappings = mappings
+        this.index = 0
+    }
+
+    _binarySearchStart (startLine, startCol, startIndex, endIndex) {
+        if (startIndex === endIndex) {
```

Comment:
I like the addition of this `sourcemapScanner` class.

This is kind of what I had in mind with regards to pulling the logic for creating multiple `CovSource ` instances into some sort of factory, and trying to keep additional logic to a minimal in that file.